app/grandchallenge/profiles/views.py

```diff
@@ -209,3 +218,18 @@ class PreSocialView(FormView):
 
     def get_success_url(self, *args, **kwargs):
         return reverse("social:begin", args=["google-oauth2"])
+
+
+class UserProfileViewSet(viewsets.ReadOnlyModelViewSet):
+    serializer_class = UserProfileSerializer
+    permission_classes = (DjangoObjectOnlyPermissions,)
+    filter_backends = (UserProfileObjectPermissionsFilter,)
+    queryset = UserProfile.objects.all()
+
+    @action(detail=False, methods=["get"])
+    def self(self, request):
```

Comment:
Do we really need the `self` view? The `view_profile` permission is only assigned to the user that owns the profile, so I think that this does the same thing as the normal list view.

Reply:
Currently the permission is indeed only assigned to the user that owns it, but I thought this might change in the future? Since right now it is possible to see profiles of other users because the permission check is not implemented in those views. Also for superusers it would not work as expected.